### DISCUSSION TITLE
How to create a dependency whitelist?

### DISCUSSION THREAD

**Discussion by db-ris-team42** [renovatebot/renovate#14713] [Request Help] [answered]
For a few older projects I want renovate to suggest dependency upgrades selectively, i.e. I want to use a whitelist of packages that may be upgraded. The rest should be ignored. This is what I came up with:

```js
  packageRules: [
    // rules for Java
    {
      matchDatasources: ["java"],
      matchPackagePrefixes: ["org.postgres"],
      enabled: true,
    },
    {
      matchDatasources: ["java"],
      matchPackagePrefixes: ["*"],
      enabled: false,
    },
  ],
```

However, the rules seem to be ineffective. Renovate still creates upgrade PRs for all packages. Any idea what I am missing?

**Comment by db-ris-team42** [comment, marked as answer]
I just find a way that both works and is crisper than my first approach:

```js
packageRules: [
    {
      matchPackagePatterns: ["*"],
      excludePackagePatterns: ["^org.postgres.*"],
      enabled: false,
    },
  ],
```

**Comment by HonkingGoose** [comment]
### Do not ignore your dependencies forever

I recommend you use the "Dependency Dashboard Approval" workflow instead of disabling Renovate for certain dependencies. 😉

When you follow this workflow, Renovate creates a new issue called "Dependency Dashboard". On this dashboard you'll see the status of all possible updates.

If you want to update one of your "ignored dependencies", click on the checkbox next to the update name from the Dependency Dashboard and Renovate will create the update.

### Config to copy/paste

Here's the config you'll need to try this out.

```json
{
  "dependencyDashboard": true,
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "excludePackagePatterns": ["^org.postgres.*"],
      "dependencyDashboardApproval": true
    }
  ]
}
```

#### Config changes

- Set `dependencyDashboard` to `true`
- Set `dependencyDashboardApproval` to `true` for all your "ignored updates"
- Drop `"enabled": false`
